src/lib.rs

```diff
@@ -626,23 +663,54 @@ impl PathMut for PathBuf {
                     ));
                 }
                 path::Component::RootDir => (), // leading "/" does nothing
-                path::Component::ParentDir => self.truncate_to_parent()?,
+                path::Component::ParentDir => self.pop_up()?,
             }
         }
 
         Ok(())
     }
 
-    fn truncate_to_parent(&mut self) -> Result<()> {
+    fn pop_up(&mut self) -> Result<()> {
+        /// Pop off the parent components and return how
+        /// many were removed.
+        fn pop_parent_components(p: &mut PathBuf) -> usize {
```

Comment:
this behavior needs to be better tested, and a slightly better architecture is called for. At the very least I need to handle the `.` case, and there might be others (like what to do with `...`?)

Reply:
As far as I know, `...` isn't special on any OS, so it shouldn't need special treatment.

Reply:
okay sounds good. I think I was thinking of `zsh` which treats it special, but that is probably just an extension of some kind.